### ISSUE TITLE
Display mode icons

### ISSUE THREAD

**Issue by TheAssassin** [TheAssassin/redflare-python#6]
@bonifarz requested displaying mode icons next to or instead of the modes in the frontend app in #2.

I am not sure what these mode icons are and where they can be found, I can't remember such icons. I'd appreciate a link to those icons, preferrably to some vector graphics such as SVG.

**Comment by bonifarz** [comment]
The mode and mutator icons are these, https://github.com/red-eclipse/textures/tree/master/modes

The good news for you: @iceflowRE has created lots of SVGs, which should be useful for you: https://github.com/red-eclipse/community/tree/master/Iceflower/graphics/textures/modes

**Comment by IceflowRE** [comment]
This may be importend as well if you want to edit them:
https://github.com/red-eclipse/community/blob/master/Iceflower/graphics/README.md

**Comment by TheAssassin** [comment]
This is definitely doable with your SVGs (I've used some of them already, see https://github.com/TheAssassin/redflare-python/tree/master/redflare/privilege-icons). As described there, the black and white parts had to be "differenced" to get the actual icon as the black parts are displayed by browsers and other viewers, and that just doesn't look good on a white background.
Apart from that I removed the borders to display the icons scaled to the line height in the CSS. These steps took about an hour for these few icons (including the "find out how to get those difference images" part).

Since the directory that @bonifarz linked to contains mutator icons, too (see #7 for more information on that), both features can be implemented as soon as these images have been edited with Inkscape. But until I find the time (the number of icons that have to be edited is pretty high), this feature can't be implemented. (Unless we find someone who wants to contribute the edited icons, since it's not too much work to implement the frontend part).

Btw: If you would like to include the already "differenced" images I created and maybe also the ones that are going to be created for these features, I'd rather use your repository as a submodule instead of keeping a copy of the SVGs in this repository. I'm anyway not sure whether you just didn't find/know about the "difference" menu entry or it had to be done like this for some of your applications :)

**Comment by IceflowRE** [comment]
Ouch, you need hours for this? :fearful:
Black & white parts are really easy (the black parts are only for hide some white parts, creating the icons was easier with this technic), you can "deleted" them in Inkscape really easy, just selected them and use the plane "exclusive or".
Borders are easy to remove too, in the document settings you can change the site size to content size with one click.

**Comment by IceflowRE** [comment]
If you can tell me, which icons you need, i can make them probably, (and if you want to wait ;D the next month i have not so much time, so probably i can do nothing/some/all).
But you only need a maximum of two or three minutes for one icon, because the most is just some clicks ^^

**Comment by TheAssassin** [comment]
Well, I already found out how to do it, I use Inkscape regularly but I didn't need this "difference" or "exclusive or" or whatever so far. But thanks for your explanation anyway.
It basically took me about 30 minutes of googling to find out where this "difference" menu entry is, and and about 30 minutes to actually apply the steps on all files. So you're right, doesn't take lots of time to do it for a single picture actually, but it sums up quickly.

To implement both features, I'd need all the files in directory that @bonifarz linked to (i.e. all the mode and mutator icons). If you find the time to do this, then thanks for your contribution. Otherwise I'll see when I find the time myself.

**Comment by bonifarz** [comment]
I'm not sure if I understood your last remark. You mention it "took ...  about 30 minutes to actually apply the steps on all files", but then you say you "need all the ... mode and mutator icons". Did you first refer to the privs icons you modified earlier?

**Comment by TheAssassin** [comment]
You are right, I was talking about the privilege icons, because these are the ones I've edited so far.
